### ISSUE TITLE
Pets being bugged on despawning/Pet becomes unuseable

### ISSUE THREAD

**Issue by Heroclassx** [gamefreedomgit/Maelstrom#5551]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (http://wowhead.com/)
[//]: # (cata-twinhead.twinstar.cz)

**Description:** 
Ff your pet as a DK dies and you abandon it and then resummon it, it will spawn with Mana (?) since Ghouls require Energy to use their Abilities (Claw, Gnaw, Leap) it simply can not use any abilities anymore, can not resummon it again aswell becasue it goes on a 2 mimute cooldown making your pet basically useless for trhe remaining fight.) 

I assume this issue is connected to following fix, has never had this issue before:
(Author: Ultranix
Message: Core/Pets: Fixed resummoning dead pets.

gamefreedomgit/Maelstrom#5530 )

**How to reproduce:** 

Have your pet die => watch your pet not despawn below your character frame just being there with 0 health => forced to right click and abandon on your pet frame before being able to summon a new pet => pet sometimes summons with mana

**How it should work:**

If your pet dies and you resummon it, it should be resummoned with the correct resource to use its spells.

**Database links:**

Screenshot pet with mana: https://cdn.discordapp.com/attachments/1049306466949017690/1198360297514217542/image.png?ex=65be9eff&is=65ac29ff&hm=2eed759746b34d6b75d830c1242f8de9ce85cf602d40ecc09b0d34fd203df870&

Recording of the bug happening: https://www.twitch.tv/videos/2037255925?t=1h58m49s (Timestamp 1h 58min 49sec in incase timestmap dosent work)

**Comment by Heroclassx** [comment]
Any news? Really frustrating bug since not only when your pet dies but also when you sac it, it will become basically unuseable in Arena, so once you are forced defensive and forced to sacrifice your pet, you will basically play a pet class without pet after that. 

Should also happend in PvE aswell but didn't test it there

**Comment by UltraNix** [comment]
Fixed.

**Comment by Heroclassx** [comment]
Actually not fixed, if pet dies and respawns it spawns with energy not mana but still dosent use any abilities.

**Comment by Heroclassx** [comment]
any update? This bug is gamebreaking since it makes the pet unuseable, can provide twitch clips/vods if that is helpful in any way

**Comment by UltraNix** [comment]
@Heroclassx Yes, please.

**Comment by Heroclassx** [comment]
https://www.twitch.tv/videos/2048144337 

So I made a highlight out of it. It is 8 minutes long but I will give timestamps in this post of what is going on where.

at the start of it I am in a solo queue game, around the 23 second mark my pet dies, I am dismissing it and resummoning it. 
After that my pet is bugged, it will not cast any abilities anymore for the rest of the game.

around 45 seconds u see me press gnaw 2-3 times weithout it going off, my pet has the energy to use it, it is in range and has no spells lined up (claw is toggled off) 

This happens throughout the arena game a few times but I don't think i Need to list it everytime here, my pet wont use any spells (huddle, claw, gnaw & leap till the end of the arena) 

I leave the arena at 1:40min, the main issue here is, the pet is still not fixed. Even after the arena the pet remains bugged, I am asking a friend to duel me (to reset the raise dead cd off 2 minutes a few times so i can resummon my pet multiple times until its eventually fixed) 

The friend which I asked for help asks what is bugged and I will from there explain it on stream exactly what happens but I have no idea why it happens, all I can say this bug has been happening since the recent pet despawn change. 

TLDR: Check the linked VOD from around 2:40ish, it will show the bug happen multiple times and me explaining on voice when it starts happening for me. Before 2:40 I have an arena game in which first the pet works, then it dies and it stops using its spell from there. The bug has been happening since the pet despawn change mid January. 

If there is any questions I can provide more info & video footage if needed. 

Thank you for the help

**Comment by Heroclassx** [comment]
at around 7:55 mark my pet starts working again after around 10-12 resummons and countless relogs

**Comment by UltraNix** [comment]
How did you revive him? I see the cooldowns are clear and it should not be the case for ghouls.
Used some macro?

**Comment by Heroclassx** [comment]
Well since the change you can not summon a pet before dismissing it anymore so I have to press /script PetAbandon() first and then /use raise dead

**Comment by UltraNix** [comment]
You can just use Raise Dead while your pet is already dead.

**Comment by Heroclassx** [comment]
I figured, I have been using a macro since 2008 retail wrath /use [nopet] raise dead and its not working on this server so I worked around it with dismissing pet which I use on classic wrath right now. But I guess this is causing the issue, seems still like a bug on whitemane but I guess can work around it

**Comment by UltraNix** [comment]
Because your are using macro to abandon your pet.

**Comment by UltraNix** [comment]
Fixed.
